relay/chunkstore/chunk_writer.go

```diff
@@ -85,3 +104,26 @@ func (c *chunkWriter) PutChunkCoefficients(
 		FragmentSizeBytes:   uint32(c.fragmentSize),
 	}, nil
 }
+
+func (c *chunkWriter) proofExists(ctx context.Context, blobKey corev2.BlobKey) bool {
```

Comment:
Can we make `proofExists` and `coefficientsExists` public so that the encoder could use them to also be idempotent?

Reply:
If the check only comes at the writing stage, then failures in the encoding manager metadata update stage will result in the blob being reencoded...

Reply:
Ah you're right. Will export these methods